### PR TITLE
GH-2948: Fix NPE when using the AvroParquetReader.Builder with LocalInputFile

### LINKED ISSUE #2948: NullPointerException when using the AvroParquetReader.Builder with a LocalInputFile
### Describe the bug, including details regarding any error messages, version, and platform.

```java
import org.apache.parquet.io.InputFile;
import org.apache.parquet.io.LocalInputFile;
import org.apache.parquet.hadoop.ParquetReader;
import org.apache.avro.generic.GenericRecord;
import org.apache.parquet.avro.AvroParquetReader;

InputFile inputFile = new LocalInputFile(Paths.get('/path/to/parquet/file');
ParquetReader<GenericRecord> reader = AvroParquetReader.<GenericRecord>builder(inputFile).build();
```

This results in a `NullPointerException` in the `org.apache.parquet.avro.AvroParquetReader.Builder#getReadSupport` method where `configuration` is still `null`, because it is not set in the `org.apache.parquet.hadoop.ParquetReader.Builder#Builder(org.apache.parquet.io.InputFile)` constructor.

Versions:
```
<dependency>
      <groupId>org.apache.hadoop</groupId>
      <artifactId>hadoop-common</artifactId>
      <version>3.4.0</version>
    </dependency>
    <dependency>
      <groupId>org.apache.hadoop</groupId>
      <artifactId>hadoop-client</artifactId>
      <version>3.4.0</version>
    </dependency>
    <dependency>
      <groupId>org.apache.parquet</groupId>
      <artifactId>parquet-avro</artifactId>
      <version>1.14.1</version>
    </dependency>
```

### Component(s)

Avro

### PR DESCRIPTION
**Pull request by joyCurry30**
### Rationale for this change
Refer to : https://github.com/apache/parquet-java/issues/2948

### What changes are included in this PR?
Fix the npe when using the AvroParquetReader.Builder with a LocalInputFile.

### Are these changes tested?
Yes

### Are there any user-facing changes?

Closes #2948

### CODE REVIEW

File: parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetReader.java [hunk 1]
```diff
@@ -230,11 +230,12 @@ protected Builder(InputFile file) {
       if (file instanceof HadoopInputFile) {
         HadoopInputFile hadoopFile = (HadoopInputFile) file;
         this.conf = hadoopFile.getConfiguration();
-        this.configuration = new HadoopParquetConfiguration(this.conf);
         optionsBuilder = HadoopReadOptions.builder(this.conf, hadoopFile.getPath());
       } else {
+        this.conf = new Configuration();
         optionsBuilder = ParquetReadOptions.builder(new HadoopParquetConfiguration());
```
Comment (Fokko) [inline parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetReader.java:236]: The idea behind `HadoopParquetConfiguration` is to run without Hadoop. With this line we initialize a Hadoop configuration while it is not a Hadoop file.

What do you think of:
```suggestion
        this.configuration = new HadoopParquetConfiguration();
        optionsBuilder = ParquetReadOptions.builder(this.configuration);
```

Comment (joyCurry30) [reply parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetReader.java:236]: I think it's better than mine.

File: parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetReader.java [hunk 2]
```diff
@@ -54,8 +54,8 @@
 @RunWith(Parameterized.class)
 public class TestParquetReader {
 
-  private static final Path FILE_V1 = createTempFile();
-  private static final Path FILE_V2 = createTempFile();
+  public static final Path FILE_V1 = createTempFile();
```
Comment (wgtmac) [inline parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetReader.java:57]: Could you please revert these lines?

**Review by wgtmac — CHANGES_REQUESTED** [review]

File: parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetReader.java (hunk 2 above)
Comment (joyCurry30) [reply parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetReader.java:57]: Ok, I will do it.

**Review by wgtmac — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**